server/tests/src/main/scala/sttp/tapir/server/tests/TestServerInterpreter.scala

```diff
@@ -17,4 +18,9 @@ trait TestServerInterpreter[F[_], +R, OPTIONS, ROUTE] {
 
   def server(routes: NonEmptyList[ROUTE]): Resource[IO, Port]
 
+  /** Exposes additional `stop` effect, which allows stopping the server inside your test. It will be called after the test anyway (assuming
+    * idempotency), but may be useful for some cases where tests need to check specific behavior like returning 503s during shutdown.
+    */
+  def serverWithStop(routes: NonEmptyList[ROUTE], gracefulShutdownTimeout: Option[FiniteDuration] = None): Resource[IO, (Port, IO[Unit])] =
```

Comment:
Maybe that's for a subsequent PR, but maybe it would make sense ot leave this as the *only* method that has to be implemented by specific test server interpreters? Then `server` can be  implemented in terms of this function, by dropping the second element of the typle